test/SILOptimizer/peephole_thick_to_objc_metatype.sil

```diff
@@ -115,3 +117,47 @@ bb0(%0 : $T):
   strong_release %0 : $T
   return %3 : $@thick T.Type
 }
+
+// CHECK-LABEL: sil @$test_peephole_objc_to_thick_to_objc
+// CHECK: [[T:%.*]] = apply
+// CHECK-NOT: {{objc_to_thick_metatype }}
+// CHECK-NOT: {{thick_to_objc_metatype }}
+// CHECK: enum $Optional<@objc_metatype AnyObject.Type>, #Optional.some!enumelt, [[T]] : $@objc_metatype AnyObject.Type
+// CHECK: return
```

Comment:
Rather than pattern checking return to match the end of the function, it is better to pattern match on the last '}' of the SIL function.

The SIL printer on purpose emits a special unique pattern on that line so you can easily pattern match:

```
// CHECK: } // end sil function 'test_peephole_objc_to_thick_to_objc'
```